src/hb-subset-plan.cc

```diff
@@ -87,42 +87,51 @@ _remap_indexes (const hb_set_t *indexes,
 #ifndef HB_NO_SUBSET_LAYOUT
 typedef void (*layout_collect_func_t) (hb_face_t *face, hb_tag_t table_tag, const hb_tag_t *scripts, const hb_tag_t *languages, const hb_tag_t *features, hb_set_t *lookup_indexes /* OUT */);
 
+template <typename T>
 static void _collect_subset_layout (hb_face_t		 *face,
 				    hb_tag_t		  table_tag,
 				    const hb_set_t	 *layout_features_to_retain,
-				    bool		  retain_all_features,
 				    layout_collect_func_t layout_collect_func,
 				    hb_set_t		 *lookup_indices /* OUT */)
 {
-  if (retain_all_features)
+  hb_blob_ptr_t<T> table = hb_sanitize_context_t ().reference_table<T> (face);
```

Comment:
You shouldn't be sanitizing yourself.  Use existing facilities to avoid duplicate work.